docs/variants/qemu_q35/recovery.md

```diff
@@ -1,29 +1,8 @@
 # Recovery
 
-The recovery process of Dasharo (UEFI) firmware on QEMU Q35 does not need any
-special actions. QEMU Q35 is an emulated platform, not real hardware, so it
-doesn't brick itself.
+The recovery process of Dasharo (coreboot+UEFI) firmware on QEMU Q35 does not
+need any special actions. QEMU Q35 is an emulated platform, not real hardware,
+so it doesn't brick itself.
 
-If something goes wrong simply rerun the QEMU command:
-
-```bash
-qemu-system-x86_64 -machine q35,smm=on \
-	-global driver=cfi.pflash01,property=secure,value=on \
-	-drive if=pflash,format=raw,unit=0,file=Build/OvmfX64/RELEASE_GCC5/FV/OVMF_CODE.fd,readonly=on \
-	-drive if=pflash,format=raw,unit=1,file=Build/OvmfX64/RELEASE_GCC5/FV/OVMF_VARS.fd \
-	-debugcon file:debug.log -global isa-debugcon.iobase=0x402 \
-	-global ICH9-LPC.disable_s3=1
-```
-
-Or build a debug image, gather the logs in `debug.log` file
-
-```bash
-qemu-system-x86_64 -machine q35,smm=on \
-	-global driver=cfi.pflash01,property=secure,value=on \
-	-drive if=pflash,format=raw,unit=0,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF_CODE.fd,readonly=on \
-	-drive if=pflash,format=raw,unit=1,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF_VARS.fd \
-	-debugcon file:debug.log -global isa-debugcon.iobase=0x402 \
-	-global ICH9-LPC.disable_s3=1
-```
-
-and report an issue with attached log.
+If something goes wrong simply rerun the QEMU command as described in
+[Dasharo initial deployment](initial-deployment.md).
```

Comment:
What about the cleanup of variables if those were set?

Reply:
`build.sh` does `make distclean`, so `coreboot.rom` will be recreated from scratch with an empty SMMSTORE.

Maybe instructions should suggest running QEMU with a copy of `build/coreboot.rom` and keeping `buid/coreboot.rom` as a backup (possibly after removing the whole `coreboot` clone), because rerunning of `build.sh` is really a waste of time (it will remove everything just to recreate it).

Reply:
Would something like `dd if=/dev/zero of=build/coreboot.rom bs=256 count=1 conv=notrunc` be acceptable? This clears SMMSTORE header, which causes `FaultTolerantWriteDxe` to reinitialize it, clearing all variables in the process.

In theory we could also move SMMSTORE to separate file, just like OVMF does. Not sure how much of coreboot build system would have to be changed, though.

Reply:
If you want to keep the image but drop the settings, I'd rather add `clear` command to `smmstoretool` which can already find `SMMSTORE` region inside a ROM.

Reply:
Good idea, added https://github.com/Dasharo/dasharo-issues/issues/893 to sort it out later and not delay the release any longer.

Reply:
https://github.com/Dasharo/docs/compare/2a47bdd5f3eae81d35cc9545010489747bac76c6..fd1fc06fa12f4d9dde6a66b01a567657c6bb1a5d

`dd` for now, may be improved later.